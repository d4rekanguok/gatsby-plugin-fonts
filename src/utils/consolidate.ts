import fs from 'fs-extra'
import { promisify } from 'util'
import path from 'path'
import glob from 'glob'

const globP = promisify(glob)

interface ConsolidateArgs {
  /* List of directories of content */
  files: string;
  root: string;
}

type Consolidate = (args: ConsolidateArgs) => Promise<string>
export const consolidate: Consolidate = async ({ root, files }) => {
  const read = async (file: string) => await fs.readFile(file, 'utf8')

  const absolutePath = path.join(root, files)
  const matchedPaths = await globP(absolutePath, { nodir: true })

  const contents = await Promise.all(matchedPaths.map(read))
  return contents.join('')
}