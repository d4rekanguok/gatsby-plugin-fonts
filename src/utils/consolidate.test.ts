import { consolidate } from './consolidate'

describe('utils/consolidate', () => {
  it('should retrieve all texts', async () => {
    const result = await consolidate({
      root: __dirname,
      files: './consolidate-test/**/*'
    })

    expect(result).toMatchSnapshot()
  })
})