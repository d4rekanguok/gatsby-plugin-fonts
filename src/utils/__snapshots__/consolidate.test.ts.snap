// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/consolidate should retrieve all texts 1`] = `
"holla, hallo한국어{
  \\"lang\\": \\"일본어\\"
}"
`;
